### ISSUE TITLE
absolute path on File rather than Relative

### ISSUE THREAD

**Issue by webmasterMeyers** [dustin10/VichUploaderBundle#1044]
### BC Break Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 1.10.2

#### Summary

Considering this entity mapping

```
    /**
     * @Assert\NotBlank()
     * @Vich\UploadableField(mapping="assets", fileNameProperty="filename")
     */
    private $file;
```
#### Previous behavior

in version `1.10.1` `$profilePhoto->getFile()->getPathname()` will return a relative path like `photos/123456.jpg`  

#### Current behavior

in version `1.10.2` `$profilePhoto->getFile()->getPathname()` will return an absolute path like `/var/www/html/example.com/public_html/photos/123456.jpg`

#### How to reproduce

1. dump the contents of a "file", note the path variables
2. update vich uploader bundle
3. dump the contents of a "file", note the path variables

**Comment by webmasterMeyers** [comment]
I am using Flysytem with both a local and S3 adaptor depending on the dev or prod environment.

I suspect maybe the [change in Storage/FlysystemStorage.php](https://github.com/dustin10/VichUploaderBundle/compare/1.10.1...1.10.2#diff-0e420f938cd72181787723ba8daea67f) is the culprit

**Comment by webmasterMeyers** [comment]
yes, reverting that single change returns the previous behaviour.  Why the change?  How else would I go about getting the relative file path?

**Comment by garak** [comment]
See discussion in #493

**Comment by webmasterMeyers** [comment]
fair enough, so the expected behaviour is to receive an absolute path then.  What is the best way to get the relative path?  or do I just hard code it?

**Comment by garak** [comment]
You should know the path of your assets, since you configured it.
Just retrieve such path from configuration

**Comment by guillaume-ro-fr** [comment]
Hi everyone !
Another problem is that the method `applyPathPrefix()` is not registred on the interface [AdapterInterface.php](https://github.com/thephpleague/flysystem/blob/master/src/AdapterInterface.php).

So, when you use the `MemoryAdapter` from Flysystem, you got an error : 
```
Call to undefined method League\Flysystem\Memory\MemoryAdapter::applyPathPrefix() 

#1 /builds/xxxx/vendor/vich/uploader-bundle/Storage/FlysystemStorage.php:64
```
